packages/twenty-server/src/database/commands/upgrade-version/0-32/0-32-migrate-phone-fields-to-phones.command.ts

```diff
@@ -0,0 +1,361 @@
+        const workspaceDataSource =
+          await this.typeORMService.connectToDataSource(dataSourceMetadata);
+
+        if (!workspaceDataSource) {
+          throw new Error(
+            `Could not connect to dataSource for workspace ${workspaceId}`,
+          );
+        }
+        const standardPersonPhoneFieldWithTextType =
+          await this.fieldMetadataRepository.findOneBy({
+            workspaceId,
+            standardId: PERSON_STANDARD_FIELD_IDS.phone,
+          });
+
+        if (!standardPersonPhoneFieldWithTextType) {
+          throw new Error(
+            `Could not find standard phone field on person for workspace ${workspaceId}`,
+          );
+        }
+
+        await this.migrateStandardPersonPhoneField({
+          standardPersonPhoneField: standardPersonPhoneFieldWithTextType,
+          workspaceDataSource,
+          workspaceSchemaName: dataSourceMetadata.schema,
+        });
+
+        const fieldsWithPhoneType = await this.fieldMetadataRepository.find({
+          where: {
+            workspaceId,
+            type: FieldMetadataType.PHONE,
+          },
+        });
+
+        for (const deprecatedPhoneField of fieldsWithPhoneType) {
+          await this.migrateCustomPhoneField({
+            phoneField: deprecatedPhoneField,
+            workspaceDataSource,
+            workspaceSchemaName: dataSourceMetadata.schema,
+          });
+        }
+      } catch (error) {
+        this.logger.log(
+          chalk.red(
+            `Field migration on workspace ${workspaceId} failed with error: ${error}`,
+          ),
+        );
+        continue;
+      }
+      this.logger.log(chalk.green(`Command completed!`));
+    }
+  }
+
+  private async migrateStandardPersonPhoneField({
+    standardPersonPhoneField,
+    workspaceDataSource,
+    workspaceSchemaName,
+  }: {
+    standardPersonPhoneField: FieldMetadataEntity;
+    workspaceDataSource: DataSource;
+    workspaceSchemaName: string;
+  }) {
+    const personObjectMetadata = await this.objectMetadataRepository.findOne({
+      where: { id: standardPersonPhoneField.objectMetadataId },
+    });
+
+    if (!personObjectMetadata) {
+      throw new Error(
+        `Could not find Person objectMetadata (id ${standardPersonPhoneField.objectMetadataId})`,
+      );
+    }
+
+    this.logger.log(`Attempting to migrate standard person phone field.`);
+    const workspaceQueryRunner = workspaceDataSource.createQueryRunner();
+
+    await workspaceQueryRunner.connect();
+    const { id: _id, ...deprecatedPhoneFieldWithoutId } =
+      standardPersonPhoneField;
+
+    const workspaceId = standardPersonPhoneField.workspaceId;
+
+    try {
+      let standardPersonPhonesFieldType =
+        await this.fieldMetadataRepository.findOneBy({
+          workspaceId,
+          standardId: PERSON_STANDARD_FIELD_IDS.phones,
+        });
+
+      if (!standardPersonPhonesFieldType) {
+        const phoneDefaultValue = deprecatedPhoneFieldWithoutId.defaultValue;
```

Comment:
Don't think we need to handle default value here, there should be none (actually empty string would be the default value if type was TEXT but nothing to really copy from)